books.php

```diff
@@ -15,6 +15,85 @@ public static function getInstance()
         }
         return self::$instance;
     }
+    public function auth($login, $password)
+    {
+        $stmt = $this->link->prepare("SELECT id FROM users WHERE BINARY login = :login AND BINARY password = :password");
+        $stmt->execute([':login' => $login, ':password' => $password]);
+        if ($stmt->rowCount() > 0) {
+            return true;
+        }
+        return false;
+    }
+    public function add($data)
+    {
+        $i = 0;
+        $arrayTable = ['name' => 0, 'genre' => 0, 'pages' => 0];
+        if ((count($data, COUNT_RECURSIVE) - count($data)) > 0) {
+            foreach ($data as $key => $array1) {
+                var_dump(implode("','", array_intersect_key($array1, $arrayTable)));
+                $stmt = $this->link->prepare("INSERT INTO `book`(`name`, `genre`, `pages`) VALUES (" . "'" . implode("','", array_intersect_key($array1, $arrayTable)) . "'" . ")");

```

Comment:
In this case you don't escape input variables before sql query.
You can use bindParam or bindParams from pdo library.